### ISSUE TITLE
feat(linter): no-restricted-imports should support dynamic import() expressions

### ISSUE THREAD

**Issue by johan-afs** [oxc-project/oxc#21411]
## Summary

The `eslint/no-restricted-imports` rule currently only applies to static imports (import declarations and `TSImportEqualsDeclaration`). It does not flag dynamic `import()` expressions with string literal arguments.

This means code like:

```ts
const { default: Anthropic } = await import('@anthropic-ai/sdk');
const { default: OpenAI } = await import('openai');
```

…is not caught, even when `@anthropic-ai/sdk` or `openai` are explicitly restricted.

## Motivation

Dynamic `import()` with a string literal is functionally equivalent to a static import — the module is known at lint time. In our codebase, developers use lazy `await import('...')` to defer loading of heavy SDKs, which bypasses the restriction entirely. There's no way to enforce "don't import this module" when both static and dynamic forms exist.

The current docs acknowledge this: *"It applies to static imports only, not dynamic ones."*

## Proposed behavior

When `import()` is called with a **string literal** argument, check it against the configured `paths` and `patterns` just like a static import. Dynamic expressions (e.g. `import(variable)`) should continue to be ignored — those are the domain of `import/no-dynamic-require`.

Since dynamic `import()` doesn't produce named bindings at the AST level (`const { foo } = await import('bar')` is destructuring, not an import binding), the `importNames` / `allowImportNames` options would not apply to dynamic imports — only path-level and pattern-level restrictions would.

### Example config

```jsonc
{
  "no-restricted-imports": ["error", {
    "paths": [{ "name": "@anthropic-ai/sdk", "message": "Use the shared LLM abstraction instead." }],
    "patterns": [{ "group": ["openai", "openai/*"] }]
  }]
}
```

**Before (not flagged):**
```ts
const { default: Anthropic } = await import('@anthropic-ai/sdk'); // no error
```

**After (flagged):**
```ts
const { default: Anthropic } = await import('@anthropic-ai/sdk');
// ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
// error: '@anthropic-ai/sdk' import is restricted. Use the shared LLM abstraction instead.
```

## Implementation notes

The change would be in `crates/oxc_linter/src/rules/eslint/no_restricted_imports.rs`:

- Add an `AstKind::ImportExpression` arm to the existing `run()` method
- Extract the source from `import_expr.source` only when it's a `StringLiteral`
- Reuse the existing path/pattern checking logic
- Skip `importNames` / `allowImportNames` checks for dynamic imports (no named bindings)
- Update the rule documentation to reflect the new coverage

Happy to submit a PR for this if the approach sounds right.

**Comment by camchenry** [comment]
Linking the ESLint discussion on this from some years ago for context: https://github.com/eslint/eslint/issues/13594.

It looks like there was discussion around this not being useful in all cases (like `await import ("@foo" + bar)`), but only in static ones. I'm fine with that really, this change seems aligned with the intent of the rule: to prevent people from importing specific modules. We can indicate in the docs that dynamically calculated module names are not supported, but this seems like a very useful extension of the rule to me.

Also we've established that new errors being added by minor release versions is okay because the analysis has improved and highlighted errors that we previously missed: https://oxc.rs/docs/guide/usage/linter/versioning.html#are-new-lint-errors-a-breaking-change
